File: PF2-web2/web2-backend/src/main/java/com/example/demo/bean/ExternalEntity.java
package com.example.demo.bean;

import java.util.ArrayList;

public class ExternalEntity extends PdNode{
	private int externalentity_no;
	private String externalentity_name;
	private String externalentity_shortname;
	private int externalentity_x;
	private int externalentity_y;
	private int externalentity_h;
	private int externalentity_w;
	private ArrayList<Phenomenon> phes;
//	private ArrayList<IntentPhenomenon> phes;
	public ExternalEntity() {
		super();
	}

	public ExternalEntity(ExternalEntity item) {
		externalentity_no = item.externalentity_no;
		externalentity_name = item.externalentity_name;
		externalentity_shortname = item.externalentity_shortname;
		externalentity_x = item.externalentity_x;
		externalentity_y = item.externalentity_y;
		externalentity_h = item.externalentity_h;
		externalentity_w = item.externalentity_w;
	}
	public void setPhes(ArrayList<Phenomenon> phes){
		this.phes = phes;
	}
	public ArrayList<Phenomenon> getPhes(){
		return this.phes;
	}
	public int getExternalentity_no() {
		return externalentity_no;
	}
	public void setExternalentity_no(int externalentity_no) {
		this.externalentity_no = externalentity_no;
	}
	public String getExternalentity_name() {
		return externalentity_name;
	}
	public void setExternalentity_name(String externalentity_name) {
		this.externalentity_name = externalentity_name;
	}
	public String getExternalentity_shortname() {
		return externalentity_shortname;
	}
	public void setExternalentity_shortname(String externalentity_shortname) {
		this.externalentity_shortname = externalentity_shortname;
	}
	public int getExternalentity_x() {
		return externalentity_x;
	}
	public void setExternalentity_x(int externalentity_x) {
		this.externalentity_x = externalentity_x;
	}
	public int getExternalentity_y() {
		return externalentity_y;
	}
	public void setExternalentity_y(int externalentity_y) {
		this.externalentity_y = externalentity_y;
	}
	public int getExternalentity_h() {
		return externalentity_h;
	}
	public void setExternalentity_h(int externalentity_h) {
		this.externalentity_h = externalentity_h;
	}
	public int getExternalentity_w() {
		return externalentity_w;
	}
	public void setExternalentity_w(int externalentity_w) {
		this.externalentity_w = externalentity_w;
	}
	public int getNo() {
		return externalentity_no;
	}

	public void setNo(int no) {
		this.externalentity_no = no;
	}

	public String getName() {
		return externalentity_name;
	}

	public void setName(String name) {
		this.externalentity_name = name;
	}

	public String getShortname() {
		return externalentity_shortname;
	}

	public void setShortname(String shortname) {
		this.externalentity_shortname = shortname;
	}

	public int getX() {
		return externalentity_x;
	}

	public void setX(int x) {
		this.externalentity_x = x;
	}

	public int getY() {
		return externalentity_y;
	}

	public void setY(int y) {
		this.externalentity_y = y;
	}

	public int getH() {
		return externalentity_h;
	}

	public void setH(int h) {
		this.externalentity_h = h;
	}

	public int getW() {
		return externalentity_w;
	}

	public void setW(int w) {
		this.externalentity_w = w;
	}
}

File: PF2-web2/web2-frontend/src/app/entity/ExternalEntity.ts
// import { IntentPhenomenon } from "./IntentPhenomenon";
import { Phenomenon } from './Phenomenon';
import { Node } from "./Node";
export class ExternalEntity extends Node {
  externalentity_no: number;
  externalentity_name: string;
  externalentity_shortname: string;
  externalentity_x: number;
  externalentity_y: number;
  externalentity_h: number;
  externalentity_w: number;
  // phes: IntentPhenomenon[];
  phes: Phenomenon[];
  static newExternalEntity(no, name, shortName, x, y, w, h) {
    let pd = new ExternalEntity();
    pd.externalentity_no = no;
    pd.externalentity_name = name;
    pd.externalentity_shortname = shortName;
    pd.externalentity_x = x;
    pd.externalentity_y = y;
    pd.externalentity_h = h;
    pd.externalentity_w = w;
    return pd;
  }

  static newExternalEntityWithOld(old, name, shortName) {
    let pd = new ExternalEntity();
    pd.externalentity_no = old.externalentity_no;
    pd.externalentity_x = old.externalentity_x;
    pd.externalentity_y = old.externalentity_y;
    pd.externalentity_h = old.externalentity_h;
    pd.externalentity_w = old.externalentity_w;

    pd.externalentity_name = name;
    pd.externalentity_shortname = shortName;
    return pd;
  }

  static copyExternalEntity(old) {
    let pd = new ExternalEntity();
    pd.externalentity_no = old.externalentity_no;
    pd.externalentity_x = old.externalentity_x;
    pd.externalentity_y = old.externalentity_y;
    pd.externalentity_h = old.externalentity_h;
    pd.externalentity_w = old.externalentity_w;

    pd.externalentity_name = old.externalentity_name;
    pd.externalentity_shortname = old.externalentity_shortname;
    pd.phes = new Array<Phenomenon>();
    if(old.phes!=null){
      // pd.phes = new Array<Phenomenon>();
      for(let p of old.phes){
        let ph=new Phenomenon();
        ph.phenomenon_no=p.phenomenon_no;
        ph.phenomenon_name=p.phenomenon_name;
        ph.phenomenon_type=p.phenomenon_type;
        ph.phenomenon_from=p.phenomenon_from;
        ph.phenomenon_to=p.phenomenon_to;
        pd.phes.push(ph);
      }
    }
    
    return pd;
  }
  getNo() {
    return this.externalentity_no;
  }
  getName() {
    return this.externalentity_name;
  }
  getShortName() {
    return this.externalentity_shortname;
  }
  getX() {
    return this.externalentity_x;
  }
  getY() {
    return this.externalentity_y;
  }
  getH() {
    return this.externalentity_h;
  }
  getW() {
    return this.externalentity_w;
  }
  setNo(no) {
    this.externalentity_no = no;
  }
  setName(name) {
    this.externalentity_name = name;
  }
  setShortName(shortName) {
    this.externalentity_shortname = shortName;
  }
  setX(x) {
    this.externalentity_x = x;
  }
  setY(y) {
    this.externalentity_y = y;
  }
  setH(h) {
    this.externalentity_h = h;
  }
  setW(w) {
    this.externalentity_w = w;
  }
}
